dissect/target/filesystems/zip.py

```diff
@@ -161,11 +161,12 @@ def lstat(self) -> fsutil.stat_result:
         elif not self.entry.is_dir() and not stat.S_ISREG(mode):
             mode = stat.S_IFREG | mode
 
-        date_time = self.entry.date_time
-        if date_time[0] < 1980 or date_time[:3] == (1980, 0, 0):
-            date_time = (1980, 1, 1, 0, 0, 0)
-        elif date_time[0] > 2107:
-            date_time = (2107, 12, 31, 23, 59, 59)
+        try:
+            timestamp = datetime(*self.entry.date_time, tzinfo=timezone.utc).timestamp()
+        except ValueError:
+            # 4354819199 ==  datetime(*(2107, 12, 31, 23, 59, 59), tzinfo=timezone.utc).timestamp()
+            # 315532800 ==  datetime(*(1980, 1, 1, 0, 0, 0), tzinfo=timezone.utc).timestamp()
+            timestamp = 4354819199 if self.entry.date_time[0] >= 2107 else 315532800
```

Comment:
```suggestion
        try:
            mtime = datetime(*self.entry.date_time, tzinfo=timezone.utc)
        except ValueError:
            mtime = datetime(*(2107, 12, 31, 23, 59, 59) if self.entry.date_time[0] >= 2107 else *(1980, 1, 1, 0, 0, 0), tzinfo=timezone.utc)
```
Maybe a little nicer than hardcoding a magic number.